src/auxiliary/t8code.jl

```diff
@@ -0,0 +1,465 @@
+"""
+    init_t8code()
+"""
+function init_t8code()
+  t8code_package_id = t8_get_package_id()
+  if t8code_package_id >= 0
+    return nothing
+  end
+
+  # Initialize `p4est` with log level ERROR to prevent a lot of output in AMR simulations
+  t8_init(SC_LP_ERROR)
+
+  return nothing
+end
+
+@inline t8_mpi_comm() = mpi_comm().val
+
+function trixi_t8_unref_forest(forest)
+  t8_forest_unref(Ref(forest))
+end
+
+function t8_free(ptr)
+  sc_free(t8_get_package_id(), ptr)
+end
+
+function finalize_t8code()
+  sc_finalize();
+  return nothing
+end
+
+function trixi_t8_count_interfaces(forest)
+  # /* Check that forest is a committed, that is valid and usable, forest. */
```

Comment:
These two-style comments look a bit weird to me

Reply:
Edited all two-style comments. These came from copying over C++ code.